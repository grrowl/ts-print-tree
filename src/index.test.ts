import { jest, describe, expect, test } from "@jest/globals";
import { ignoredPatterns, tree, VisibilityLevel } from ".";

describe("tree module", () => {
  test("calls tree() and matches console.log snapshot", () => {
    console.log = jest.fn();

    tree();
    expect((console.log as jest.Mock).mock.calls).toMatchSnapshot();
  });

  test("calls tree() with custom ignored patterns and matches console.log snapshot", () => {
    console.log = jest.fn();
    const customIgnored = ["node_modules", /\.git/, /^src\/(?!tests)/];

    tree(process.cwd(), customIgnored);
    expect((console.log as jest.Mock).mock.calls).toMatchSnapshot();
  });

  test("calls tree() with custom visibility level and matches console.log snapshot", () => {
    console.log = jest.fn();
    const customVisibilityLevel = VisibilityLevel.Private;

    tree(process.cwd(), ignoredPatterns, customVisibilityLevel);
    expect((console.log as jest.Mock).mock.calls).toMatchSnapshot();
  });
});
