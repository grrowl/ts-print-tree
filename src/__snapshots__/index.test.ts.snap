// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tree module calls tree() and matches console.log snapshot 1`] = `
[
  [
    "└─ src/
   ├─ cli.ts
   ├─ index.ts
   │  └─ const ignoredPatterns
   │  └─ function tree(rootDir: string, ignored: (string | RegExp)[], visibilityLevel: VisibilityLevel): void
   └─ tests/
      ├─ class.ts
      │  └─ export default class DefaultClass
      │  └─ class NamedClass
      └─ consts.ts
         └─ const publicArray
         └─ interface PublicInterface
         └─ const publicFunction
",
  ],
]
`;

exports[`tree module calls tree() with custom ignored patterns and matches console.log snapshot 1`] = `
[
  [
    "└─ src/
   └─ tests/
      ├─ class.ts
      │  └─ export default class DefaultClass
      │  └─ class NamedClass
      └─ consts.ts
         └─ const publicArray
         └─ interface PublicInterface
         └─ const publicFunction
",
  ],
]
`;

exports[`tree module calls tree() with custom visibility level and matches console.log snapshot 1`] = `
[
  [
    "└─ src/
   ├─ cli.ts
   │  └─ private const args
   ├─ index.ts
   │  └─ const ignoredPatterns
   │  └─ private function readTsConfig(rootDir: string): ParsedCommandLine
   │  └─ private function createProgram(rootDir: string): Program
   │  └─ private function analyzeFile(sourceFile: SourceFile, typeChecker: TypeChecker, visibilityLevel: VisibilityLevel): string
   │  └─ private function traverseDirectory(dir: string, program: Program, shouldIgnore: (path: string) => boolean, visibilityLevel: VisibilityLevel, prefix: string): string
   │  └─ private function shouldIgnore(ignoredPatterns: (string | RegExp)[], pathToCheck: string): boolean
   │  └─ function tree(rootDir: string, ignored: (string | RegExp)[], visibilityLevel: VisibilityLevel): void
   └─ tests/
      ├─ class.ts
      │  └─ private const tree
      │  └─ export default class DefaultClass
      │     ├─ private constructor(baz: number[]): void
      │  └─ class NamedClass
      │     ├─ private constructor(): void
      │     ├─ private double(): void
      │     ├─ private triple(): void
      │     ├─ protected quadruple(): void
      │     ├─ private octuple(): void
      └─ consts.ts
         └─ private const privateString
         └─ private const privateNumber
         └─ private const privateBoolean
         └─ const publicArray
         └─ interface PublicInterface
         └─ private const privateFunction
         └─ const publicFunction
",
  ],
]
`;
